MinStack.java

```diff
@@ -0,0 +1,59 @@
+// Time Complexity :  O(1)
+// Space Complexity : O(N)  N-> Number of elements
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
+ 
+import java.util.Stack;
+
+public class MinStack{
+	
+	/*declare two stacks stack and minstack*/
+
+    Stack<Integer> stack = new Stack<Integer>();
+    Stack<Integer> minStack = new Stack<Integer>();
+
+    public MinStack() {        
+    }
+    
+    public void push(int x) {
+    	//pushing all the elements into stack
+    	//pushing into minstack only if the new element is smaller than the previous element or if it's empty
+        stack.push(x);
+        if(minStack.isEmpty() || x<=minStack.peek()){
+            minStack.push(x);
+        }
+    }
+    
+    public int pop() {
+    	//if the top of the minstack equals the 
+        if(stack.peek() == minStack.peek()){
+           return minStack.pop();
```

Comment:
Don't use return here, you would also have to pop from the main stack and this won't let the control flow reach the stack.pop() line.